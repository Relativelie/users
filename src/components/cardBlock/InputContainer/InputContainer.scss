.cardBlock__inputContainer {
    display: flex;
    flex-direction: column;
    gap: 10px;
    margin-bottom: 10px;
    padding: 20px 12px;
    border: 1px solid #F3F3F3;
    border-radius: 8px;
}