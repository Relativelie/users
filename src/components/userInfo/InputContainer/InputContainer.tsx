import { FC } from 'react';
import { content } from '../../../content';
import { useTypedSelector } from '../../../hooks/useTypedSelector';
import { InputField } from './inputField/InputField';
import { userInfoContent } from '../UserInfoContent';

type Props = {
    catchInputValueChange: Function,
};

export const InputContainer:FC<Props> = ({ catchInputValueChange }) => {
    const { openedCardId } = useTypedSelector(
        (listsOfUsersState) => listsOfUsersState.listOfUsersReducer,
    );

    const findFilledValue = (card: any, inputName: string) => {
        let result = '';
        if (inputName.indexOf(' ') !== -1) {
            const newInputName = inputName.split(' ');
            Object.keys(card).forEach((key) => {
                if (key === newInputName[0]) {
                    Object.keys(card[key]).forEach((secondKey) => {
                        if (secondKey === newInputName[1]) {
                            result = card[key][secondKey];
                        }
                    });
                }
            });
        } else {
            Object.keys(card).forEach((key) => {
                if (key === inputName) {
                    result = card[key];
                }
            });
        }
        return result;
    };

    return (
        <div className="userInfo__form">
            {userInfoContent.map((inputField) => {
                const { id, name, type, required, resPath } = inputField;
                const openedCardContent = openedCardId !== null ? content[openedCardId - 1] : '';
                const filledValue = findFilledValue(openedCardContent, resPath);
                return (
                    <InputField
                        key={`inputField-${id}`}
                        labelText={name}
                        required={required}
                        type={type}
                        filledValue={filledValue}
                        catchInputValueChange={catchInputValueChange}
                    />
                );
            })}
        </div>
    );
};
