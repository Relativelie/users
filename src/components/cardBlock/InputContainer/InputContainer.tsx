import { FC } from 'react';
import { useTypedSelector } from '../../../hooks/useTypedSelector';
import { InputField } from './inputField/InputField';
import { listOfInputs } from '../../../data/listOfInputs';
import { data } from '../../../data/data';

import './InputContainer.scss';

type Props = {
    catchInputValueChange: Function,
};

export const InputContainer:FC<Props> = ({ catchInputValueChange }) => {
    const { openedCardId } = useTypedSelector(
        (listBlockState) => listBlockState.listBlockReducer,
    );

    const findFilledValue = (card: any, inputName: string) => {
        let result = '';
        if (inputName.indexOf(' ') !== -1) {
            const newInputName = inputName.split(' ');
            Object.keys(card).forEach((key) => {
                if (key === newInputName[0]) {
                    Object.keys(card[key]).forEach((secondKey) => {
                        if (secondKey === newInputName[1]) {
                            result = card[key][secondKey];
                        }
                    });
                }
            });
        } else {
            Object.keys(card).forEach((key) => {
                if (key === inputName) {
                    result = card[key];
                }
            });
        }
        return result;
    };

    return (
        <div className="cardBlock__inputContainer">
            {listOfInputs.map((inputField) => {
                const { id, name, type, required, resPath } = inputField;
                const openedCardContent = openedCardId !== null ? data[openedCardId - 1] : '';
                const filledValue = findFilledValue(openedCardContent, resPath);
                return (
                    <InputField
                        key={`inputField-${id}`}
                        labelText={name}
                        required={required}
                        type={type}
                        filledValue={filledValue}
                        catchInputValueChange={catchInputValueChange}
                    />
                );
            })}
        </div>
    );
};
