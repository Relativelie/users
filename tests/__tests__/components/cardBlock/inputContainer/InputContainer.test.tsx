import { InputContainer } from '../../../../../src/components/cardBlock/InputContainer/InputContainer';
import { render } from '../../../../testsSetup/test-utils';

let addListFunc: Function;
beforeEach(() => {
    addListFunc = jest.fn();
});

describe('Input container component', () => {
    test('snapshot - input container component', () => {
        const tree = render(<InputContainer catchInputValueChange={addListFunc} />);
        expect(tree).toMatchSnapshot();
    });
});
