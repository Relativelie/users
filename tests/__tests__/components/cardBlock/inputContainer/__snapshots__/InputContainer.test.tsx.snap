// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input container component snapshot - input container component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardBlock__inputContainer"
      >
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="Name"
          >
            Name
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="name"
            disabled=""
            placeholder="Name"
            type="text"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="User name"
          >
            User name
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="user name"
            disabled=""
            placeholder="User name"
            type="text"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="E-mail"
          >
            E-mail
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="e-mail"
            disabled=""
            placeholder="E-mail"
            type="email"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="Street"
          >
            Street
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="street"
            disabled=""
            placeholder="Street"
            type="text"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="City"
          >
            City
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="city"
            disabled=""
            placeholder="City"
            type="text"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="Phone"
          >
            Phone
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="phone"
            disabled=""
            placeholder="Phone"
            type="tel"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="Website"
          >
            Website
          </label>
          <input
            class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
            data-inputname="website"
            disabled=""
            placeholder="Website"
            type="url"
          />
        </div>
        <div
          class="inputFieldContainer"
        >
          <label
            class="inputFieldContainer__label"
            for="Comment"
          >
            Comment
          </label>
          <input
            class="inputFieldContainer__input inputFieldContainer__input_comment"
            data-inputname="comment"
            disabled=""
            placeholder=""
            type="text"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardBlock__inputContainer"
    >
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="Name"
        >
          Name
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="name"
          disabled=""
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="User name"
        >
          User name
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="user name"
          disabled=""
          placeholder="User name"
          type="text"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="E-mail"
        >
          E-mail
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="e-mail"
          disabled=""
          placeholder="E-mail"
          type="email"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="Street"
        >
          Street
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="street"
          disabled=""
          placeholder="Street"
          type="text"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="City"
        >
          City
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="city"
          disabled=""
          placeholder="City"
          type="text"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="Phone"
        >
          Phone
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="phone"
          disabled=""
          placeholder="Phone"
          type="tel"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="Website"
        >
          Website
        </label>
        <input
          class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
          data-inputname="website"
          disabled=""
          placeholder="Website"
          type="url"
        />
      </div>
      <div
        class="inputFieldContainer"
      >
        <label
          class="inputFieldContainer__label"
          for="Comment"
        >
          Comment
        </label>
        <input
          class="inputFieldContainer__input inputFieldContainer__input_comment"
          data-inputname="comment"
          disabled=""
          placeholder=""
          type="text"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
